flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java

```diff
@@ -220,9 +220,7 @@ void releaseBuffer() {
 
 		@Override
 		ByteBuf write(ByteBufAllocator allocator) throws IOException {
-			if (null == buffer) {
-				throw new NullPointerException();
-			}
+			Preconditions.checkNotNull(buffer, "When deserializing the buffer should have to be requested first.");
```

Comment:
Actually, the message should rather be `No buffer instance to serialize.` since the constructor leading to this code path is only used for de-serialization.